src/Dropdown.php

```diff
@@ -1613,11 +1611,58 @@ public static function showItemType($types = '', $options = [])
                 'display'             => $params['display'],
                 'rand'                => $params['rand'],
                 'track_changes'       => $params['track_changes'],
+                'init'                => $params['init'],
+                'width'               => $params['width'],
+                'aria_label'          => $params['aria_label'],
             ]);
         }
         return 0;
     }
 
+    /**
+     * Build dropdown options and check rights if needed
+     *
+     * This method dynamically builds dropdown options based on the provided types.
+     * It supports both single and multiple types (nested) and checks access rights if required.
+     *
+     * @param array|string $types The types for which to build dropdown options. Can be a single type or an array of types.
+     * @param bool         $checkright Whether to check access rights for the type(s).
+     * @return array|string|null The built dropdown options, or null if access is denied or type is invalid.
+     */
+    public static function buildDropdownOptions($types, $checkright = false): array|string|null
```

Comment:
This method seems dedicated to itemtype dropdowns, it should be reflected in its name.

```suggestion
    public static function buildItemtypesDropdownOptions($types, $checkright = false): array|string|null
```

Also, please add type hinting on the method parameters.